workflows/rnaaln.nf

```diff
@@ -0,0 +1,846 @@
+        .combine(STAGE_INPUT.out.meta_analysis)
+        .map{
+            meta,txt,upRdpc,metaB,analysis ->
+            [
+                [
+                    id:"${meta.study_id}.${meta.patient}.${meta.sample}.${meta.experiment}",
+                    patient:"${meta.patient}",
+                    sex:"${meta.sex}",
+                    sample:"${meta.sample}",
+                    read_group:"${meta.read_group}",
+                    data_type:"${meta.data_type}",
+                    date : "${meta.date}",
+                    genome_build: "${params.genome_build}",
+                    genome_annotation: "${params.genome_annotation}",
+                    read_groups_count: "${meta.numLanes}",
+                    study_id : "${meta.study_id}",
+                    date :"${new Date().format("yyyyMMdd")}",
+                    upRdpc : upRdpc
+                ],txt,analysis
+            ]
+        }.branch{
+            upload : it[0].upRdpc
+        }
+        .set{ch_h_novel_splice_payload}
+
+        // Make payload - splice junctions
+        PAYLOAD_SPLICE_JUNCTION_H(  // [val (meta), [path(cram),path(crai)],path(analysis_json)]
+            ch_h_novel_splice_payload.upload,
+            Channel.empty()
+            .mix(STAGE_INPUT.out.versions)
+            .mix(HISAT2_ALIGN.out.versions)
+            .collectFile(name: 'collated_versions.yml')
+        )
+        ch_versions = ch_versions.mix(PAYLOAD_SPLICE_JUNCTION_H.out.versions)
+
+        // Upload files - aligment
+        UPLOAD_NOVEL_SPLICE_H(PAYLOAD_SPLICE_JUNCTION_H.out.payload_files) // [val(meta), path("*.payload.json"), [path(CRAM),path(CRAI)]
+        ch_versions = ch_versions.mix(UPLOAD_NOVEL_SPLICE_H.out.versions)
+
+        // QC Matrics \\
+        // Samtools stats
+        SAMTOOLS_STATS_H(
+            MERG_DUP_H.out.cram_alignment_index,
+            Channel.fromPath(params.reference_fasta).map{ it -> [ [ id:'fasta' ], it ] }
+        )
+        ch_versions = ch_versions.mix(SAMTOOLS_STATS_H.out.versions)
+
+        // Picard
+        PICARD_COLLECTRNASEQMETRICS_H(
+            MERG_DUP_H.out.cram_alignment_index,
+            Channel.fromPath(params.ref_flat),
+            Channel.fromPath(params.reference_fasta),
+            Channel.fromPath(params.reference_fai),
+            Channel.fromPath(params.rrna_intervals)
+        )
+
+        ch_versions = ch_versions.mix(PICARD_COLLECTRNASEQMETRICS_H.out.versions)
+
+        // MultiQC
+        ch_reports = (
+            Channel.empty()
+            .mix(PICARD_COLLECTRNASEQMETRICS_H.out.metrics)
+            .mix(HISAT2_ALIGN.out.summary)
+            .mix(HISAT2_ALIGN.out.metrix)
+            .mix(SAMTOOLS_STATS_H.out.stats)
+        )
+
+        ch_multiqc = Channel.empty()
+        ch_multiqc = ch_multiqc.mix(ch_reports.collect{meta, report -> report}).ifEmpty([])
+
+        ch_multiqc_config = Channel.fromPath("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
```

Comment:
For defining variable, better to declare all files at the start of workflow. Easier management and readability